### DISCUSSION TITLE
Use a JSON encoded string  for data-glightbox

### DISCUSSION THREAD

**Discussion by hudri** [biati-digital/glightbox#412] [Ideas]
Glightbox currently can be configured by specifiying a `data-glightbox="strings_separated_with_semicolons"`. This feature is great, but the data format of a semicolon-separated string can be problematic when dealing with unknown data sourced from a database: What if an editor specified a description containing a semi-colon itself, or a quote (and which quote: single or double?)

Suggestion:  
Use a stringified JSON object for data-glightbox instead. E.g. [SplideJS](https://splidejs.com/guides/options/#by-data-attribute), [Masonry](https://masonry.desandro.com/#initialize-in-html). String interpretations are always error-prone, being able to do JSON.parse / JSON.stringify (or a server-side language like e.g. PHP's json_enode / json_decode) to create the config object solves a ton of edge cases at quite low cost.

As a side effect, this change would allow to define nested structures like e.g. [elements ](https://github.com/biati-digital/glightbox#lightbox-options) to be configured using a declarative data-glightbox syntax.

This proposal is break in backwards compatibility and would require a new major version number.

**Comment by gingerchew** [comment]
I like this idea, I've always been a fan of how Masonry handled their options like that.
